hc-nested-list-behavior.html

```diff
@@ -62,6 +62,20 @@
       'item-open-changed': 'itemOpenChanged'
     },
     /**
+     * will close each expanded tree node.
+     * Relies upon logic being in place to prevent two sibling nodes from being open concurrently.
+     *
+     * @param {Object} [parts = this.parts] - object containing all properties of each node
+     */
+    collapseTree(parts = this.parts) {
+      const children = Object.keys(parts);
+      children.forEach(child => {
```

Comment:
no parts is an object that contains a key value pair for each level of the tree.  0, 0.0, 0.0.0, 1, 1.1, 1.1.1 etc.  So we can loop through that to get to each level of the tree without a recursive call.